libs/particles/effects.ts

```diff
@@ -22,6 +23,19 @@ namespace particles {
         }
     }
 
+    /**
+     * Removes all effects at anchor's location
+     * @param anchor the anchor to remove effects from
+     */
+    //% blockId=particlesremoveeffect block="remove effects on %anchor=variables_get(mySprite)"
+    //% group="Effects"
+    export function removeEffects(anchor: ParticleAnchor) {
+        const sources = game.currentScene().data.particleSources as particles.ParticleSource[];
+        if (!sources) return;
+        sources.filter(ps => ps.anchor == anchor || ps.anchor.x == anchor.x && ps.anchor.y == anchor.y)
+                .forEach(ps => ps.destroy());
```

Comment:
remove destroyed source from sources?

Reply:
The `ParticleSource` removes itself from sources when it is destroyed after the final particle from the source is destroyed in ``prune``, so that the existing particles can finish off their lifespan in peace; I'll add a quick note in destroy to indicate this.